var/spack/repos/builtin/packages/py-dlio-profiler/package.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class PyDlioProfiler(PythonPackage):
+    """A low-level profiler for capture I/O calls from deep learning applications."""
+
+    homepage = "https://github.com/hariharan-devarajan/dlio-profiler.git"
+    git = "https://github.com/hariharan-devarajan/dlio-profiler.git"
+    maintainers("hariharan-devarajan")
+
+    version("develop", branch="dev")
+    version("master", branch="master")
+    version("0.0.1", tag="v0.0.1")
+    depends_on("cpp-logger@0.0.1")
+    depends_on("brahma@0.0.1")
+    depends_on("gotcha@develop")
+    depends_on("py-setuptools", type="build")
+    depends_on("py-pybind11", type="build")
+
+    def setup_build_environment(self, env):
+        prefix = self.spec["py-dlio-profiler"].prefix + "/local"
+        site_packages = "python{0}/dist-packages".format(self.spec["python"].version[:-1])
+        env.set("DLIO_PROFILER_DIR", prefix)
+        env.set("DLIO_PYTHON_SITE", site_packages)
```

Comment:
```suggestion
        env.set("DLIO_PROFILER_DIR", self.prefix)
        env.set("DLIO_PYTHON_SITE", python_platlib)
```
The above code only works on Debian and only when using the system Python, it doesn't work the other 99% of the time.

Reply:
i changed it based on your suggestion. For future reference what are variables present for us to do in packages? and can you point me to the documentation.

Reply:
Documentation is minimal for this. For packages that depend on Python, you have access to `python_platlib` (not pure-Python) and `python_purelib` (pure-Python) which store the site packages directory. There are also vars for the include directory.

Reply:
Yes i didnt find the documentation so I looked at the PythonPackage itself https://github.com/spack/spack/blob/a67455707a0e6f8bee44b70b4bcff156a83d465a/lib/spack/spack/build_systems/python.py#L232 using which I came up with that logic.

Reply:
BTW i need the site-packages or dist-packages within the lib not the lib itself. I should use python_purelib then right?

Reply:
On the majority of systems, `python_purelib` and `python_platlib` are the same. On some RHEL systems when using the system Python, `python_purelib` uses `lib` and `python_platlib` uses lib64. So it depends on whether or not the library is pure-Python or not.

Reply:
So both of them point to site-packages just change lib or lib64?

Reply:
Yes, `site-packages` (or `dist-packages` on some OSes) is the only location where Python libs are installed.